source/particles/data_out.cc

```diff
@@ -16,17 +16,57 @@
 #include <deal.II/particles/data_out.h>
 #include <deal.II/particles/particle_handler.h>
 
+// We use some exceptions declared in this header
+#include <deal.II/numerics/data_out_dof_data.h>
+
 DEAL_II_NAMESPACE_OPEN
 
 namespace Particles
 {
   template <int dim, int spacedim>
   void
   DataOut<dim, spacedim>::build_patches(
-    const Particles::ParticleHandler<dim, spacedim> &particles)
+    const Particles::ParticleHandler<dim, spacedim> &particles,
+    const std::vector<std::string> &                 data_component_names,
+    const std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      &data_component_interpretations_)
   {
+    Assert(
+      data_component_names.size() == data_component_interpretations_.size(),
+      ExcMessage(
+        "When calling Particles::DataOut::build_patches with data component "
+        "names and interpretations you need to provide as many data component "
+        "names as interpretations. Provide the same name for components that "
+        "belong to a single vector or tensor."));
+
+    if (data_component_names.size() > 0)
+      {
+        Assert(
+          data_component_names.size() ==
+            particles.begin()->get_properties().size(),
+          ExcMessage(
+            "When calling Particles::DataOut::build_patches with data component "
+            "names and interpretations you need to provide as many data component "
+            "names as the particles have properties."));
+      }
+
     dataset_names.clear();
     dataset_names.emplace_back("id");
+    dataset_names.insert(dataset_names.end(),
+                         data_component_names.begin(),
+                         data_component_names.end());
+
+    data_component_interpretations.clear();
+    data_component_interpretations.emplace_back(
+      DataComponentInterpretation::component_is_scalar);
+    data_component_interpretations.insert(
+      data_component_interpretations.end(),
+      data_component_interpretations_.begin(),
+      data_component_interpretations_.end());
```

Comment:
Hm, I guess my suggestion didn't actually make it more readable...

But at least it's faster this way! ;-)